AsyncDisplayKit/Details/_ASDisplayView.mm

```diff
@@ -375,3 +376,81 @@ - (UIView *)preferredFocusedView
 }
 #endif
 @end
+
+
+#pragma mark - Accessibility
+
+static const char *ASDisplayNodeAssociatedNodeKey = "ASAssociatedNode";
+
+@implementation UIAccessibilityElement (_ASDisplayView)
+
+- (void)setAsyncdisplaykit_node:(ASDisplayNode *)node
+{
+  objc_setAssociatedObject(self, ASDisplayNodeAssociatedNodeKey, node, OBJC_ASSOCIATION_ASSIGN); // Weak reference to avoid cycle, since the node retains the layer.
+  
+  // Update UIAccessibilityElement properties from node
+  self.accessibilityLabel = node.accessibilityLabel;
+  self.accessibilityHint = node.accessibilityHint;
+  self.accessibilityValue = node.accessibilityValue;
+  self.accessibilityTraits = node.accessibilityTraits;
+}
+
+- (ASDisplayNode *)asyncdisplaykit_node
+{
+  return objc_getAssociatedObject(self, ASDisplayNodeAssociatedNodeKey);
+}
+
+@end
+
+@implementation _ASDisplayView (UIAccessibilityContainer)
+
+#pragma mark - UIAccessibility
+
+- (NSArray *)accessibleElements
+{
+  ASDisplayNode *asyncDisplayKitNode = self.asyncdisplaykit_node;
+  if ( _accessibleElements != nil ) {
+    return _accessibleElements;
```

Comment:
How does this get invalidated when the node hierarchy changes?

I'd pretty strongly prefer we do no caching for now, as a) I doubt this method is called frequently, and b) it is very important for both testing and accessibility that we don't have any bugs with this information not getting invalidated.  Particularly since it is a feature that is not tested as frequently, we can layer in optimizations later if needed.